cdap-api/src/main/java/io/cdap/cdap/api/security/credential/CredentialProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.security.credential;
```

Comment:
how come this is in cdap-api? Is it because we're planning on having some of the program contexts (SparkContext, WorkflowContext, etc) extend this interface?

Reply:
Yup

Reply:
eventual meaning something in 6.10, or sometime in a future release?

If it's not in this minor release, would be better to keep it internal rather than in the api.

Reply:
Done, moved it all back to `cdap-proto`. Thanks!